js/kucoinfutures.js

```diff
@@ -0,0 +1,1287 @@
+        //                 ["5000.0", 1000],   //Price, quantity
+        //                 ["6000.0", 1983]    //Price, quantity
+        //           ],
+        //           "bids": [
+        //                 ["3200.0", 800],    //Price, quantity
+        //                 ["3100.0", 100]     //Price, quantity
+        //           ],
+        //           "ts": 1604643655040584408  // timestamp
+        //         }
+        //     }
+        //
+        const data = this.safeValue (response, 'data', {});
+        const timestamp = parseInt (Precise.stringDiv (this.safeString (data, 'ts'), '1000000'));
+        const orderbook = this.parseOrderBook (data, symbol, timestamp, 'bids', 'asks', 0, 1);
+        orderbook['nonce'] = this.safeInteger (data, 'sequence');
+        return orderbook;
+    }
+
+    async fetchL3OrderBook (symbol, limit = undefined, params = {}) {
+        throw new BadRequest (this.id + ' only can only fetch the L2 order book');
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        const response = await this.futuresPublicGetTicker (this.extend (request, params));
+        //
+        //    {
+        //        "code": "200000",
+        //        "data": {
+        //            "sequence": 1638444978558,
+        //            "symbol": "ETHUSDTM",
+        //            "side": "sell",
+        //            "size": 4,
+        //            "price": "4229.35",
+        //            "bestBidSize": 2160,
+        //            "bestBidPrice": "4229.0",
+        //            "bestAskPrice": "4229.05",
+        //            "tradeId": "61aaa8b777a0c43055fe4851",
+        //            "ts": 1638574296209786785,
+        //            "bestAskSize": 36,
+        //        }
+        //    }
+        //
+        return this.parseTicker (response['data'], market);
+    }
+
+    parseTicker (ticker, market = undefined) {
+        //
+        //     {
+        //         "code": "200000",
+        //         "data": {
+        //             "sequence":  1629930362547,
+        //             "symbol": "ETHUSDTM",
+        //             "side": "buy",
+        //             "size":  130,
+        //             "price": "4724.7",
+        //             "bestBidSize":  5,
+        //             "bestBidPrice": "4724.6",
+        //             "bestAskPrice": "4724.65",
+        //             "tradeId": "618d2a5a77a0c4431d2335f4",
+        //             "ts":  1636641371963227600,
+        //             "bestAskSize":  1789
+        //          }
+        //     }
+        //
+        const last = this.safeNumber (ticker, 'price');
+        const marketId = this.safeString (ticker, 'symbol');
+        market = this.safeMarket (marketId, market, '-');
+        const timestamp = Precise.stringDiv (this.safeString (ticker, 'ts'), '1000000');
```

Comment:
also here the timestamp should be an integer

Reply:
This one definitely needs to be divided, which is why I used `safeString`